homeassistant/components/tolo/number.py

```diff
@@ -0,0 +1,129 @@
+"""TOLO Sauna number controls."""
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import ToloSaunaCoordinatorEntity, ToloSaunaUpdateCoordinator
+from .const import DOMAIN, FAN_TIMER_MAX, POWER_TIMER_MAX, SALT_BATH_TIMER_MAX
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up number controls for TOLO Sauna."""
+    coordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        [
+            ToloPowerTimerDuration(coordinator, entry),
+            ToloSaltBathTimerDuration(coordinator, entry),
+            ToloFanTimerDuration(coordinator, entry),
```

Comment:
It would be better to use `NumberEntityDescription`s instead. It would allow you to reduce this to a single class.

Reply:
Like this? Still three classes... (`ToloNumberEntityDescriptionBase`, `ToloNumberEntityDescription` and `ToloNumberEntity`). Got some inspiration how to do this from `goodwe` integration, never worked with `EntityDescription` before. Will also consider that for other parts of the `tolo` integration if you approve the way I did it here.

To prevent questions: There will come a fourth number entity, which has a different `unit_of_measurement` (`hours`) and `max_value` (`8`) which I couldn't test yet. Therefore, these values are not set in the `ToloNumberEntityDescription` class.